### ISSUE TITLE
Unable to connect to queue with custom Visibility Timeout

### ISSUE THREAD

**Issue by kijasek** [skyscreamer/nevado#48]
Hi,

I am facing the following problem using nevado. I know aws properties of some SQS environment and queue name to which I would like to connect. However that queue has default Visibility Timeout set to 2 hours. When trying to connect I have QueueNameExists Exception with message that VisibilityTimeout property does not match.

Is it possible to connect to such queue with nevado?

Best regards,
Tomasz Kijas

**Comment by carterpage** [comment]
Well, Nevado is a whole JMS framework, it's not a simple SQS connector.  If
we resolve this issue, your next problem is going to be that Nevado expects
the messages to be serialized in with its own format.

Assuming that's the case, I think you'd have better luck with using direct
SQS access from the Amazon AWS SDK.

**Comment by kijasek** [comment]
Thanks to Nevado we have managed to switch from ActiveMQ to SQS smoothly. The only problem we had was with transactions and default visibilityTimeout (30 seconds). Our JMS consumer needs more time than 30 seconds to process message and commit transaction. That collides with SQS trying to redeliver message after visibilityTimeout has passed. So we had changed the default visibilityTimeout in SQS console, but then we couldn't connect to the queue.

I have briefly scanned Nevado source code and thought about adding some optional attributes to NevadoQueue class. And then in getSQSQueueImpl method in AmazonAwsSQSConnector class just use these properties when calling createQueue method from Amazon's SDK?

Anyway. We already have some workaround in our project and I understand your point of changing just one SQS specific option. So I am closing this issue :)

Best regards.
